### ISSUE TITLE
QGIS fails to load due to missing libspatialindex.so.7

### ISSUE THREAD

**Issue by sy137828** [qgis/QGIS#58077]
## Description:
I encountered an issue when trying to run a fresh installation of QGIS on Artix Linux. The error message was:
```
qgis: error while loading shared libraries: libspatialindex.so.7: cannot open shared object file: No such file or directory
```

## Steps to Reproduce:
1. Install QGIS on Artix Linux using ```sudo pacman -S qgis```.
2. Attempt to run QGIS.

### Versions

<html>
<body>
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /><style type="text/css">
p, li { white-space: pre-wrap; }
</style></head><body>

QGIS version | 3.38.0-Grenoble | QGIS code branch | Release 3.38
-- | -- | -- | --
Qt version | 5.15.14
Python version | 3.12.4
GDAL/OGR version | 3.9.0
PROJ version | 9.4.1
EPSG Registry database version | v11.006 (2024-03-13)
GEOS version | 3.12.2-CAPI-1.18.2
SQLite version | 3.46.0
Compiled against PDAL | 2.7.1 | Running against PDAL | 2.7.2
PostgreSQL client version | 16.2
SpatiaLite version | 5.1.0
QWT version | 6.3.0
QScintilla2 version | 2.14.1
OS version | Artix Linux
  |   |   |  
Active Python plugins
db_manager | 0.1.20

</body></html>
</body>
</html>

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

## Workaround:
I resolved the issue by creating a symbolic link to `libspatialindex.so.6`:
```bash
sudo ln -s /usr/lib/libspatialindex.so.6 /usr/lib/libspatialindex.so.7
```
## Details:

- The `spatialindex` package installed on Artix Linux provides `libspatialindex.so.6`, but QGIS seems to require `libspatialindex.so.7`.
- Creating a symbolic link from `libspatialindex.so.7` to `libspatialindex.so.6` allowed QGIS to run successfully.

## Suggestion:
Please consider updating the dependencies or providing guidance in the documentation for handling this issue. Ensuring compatibility with the available versions of `libspatialindex` on different distributions would be very helpful.

## Environment:
OS: Artix Linux
QGIS Version: 3.38.0-Grenoble

Thank you for your attention to this matter.

**Comment by nicogodet** [comment]
This is likely a packing issue rather than a QGIS issue.

Look for Artix issue tracker and report your issue.

**Comment by sy137828** [comment]
Hi,

Thanks for your response. After digging into the problem, I discovered that QGIS is actually not available in Artix Linux repositories. Since I manually added the `extra` and `multilib` repositories from Arch Linux to `/etc/pacman.conf`, my package manager basically grabbed the QGIS package from the Arch Linux repositories.

I successfully installed QGIS in an Arch Linux VM, which leads me to believe that the issue may be due to a version mismatch between the dependency packages available in the Artix Linux and Arch Linux repositories. This mismatch likely caused the `libspatialindex.so` error I encountered.

To resolve dependency issues on Artix Linux, users might need to either build the QGIS package and its dependencies locally or use the workaround I employed to get the QGIS package from the Arch Linux repository running.

I guess this issue can be closed now.
